### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by Juanse7793**
in this PR we fullfilled the following requirements:

## Requirements

- Add unit tests to your classes: :100: 
  - `Book` :heavy_check_mark: 
  - `Person` :heavy_check_mark: 
  - `Student` :heavy_check_mark: 
  - `Teacher` :heavy_check_mark: 
  - `Classroom` :heavy_check_mark: 
  - `Rental` :heavy_check_mark:  
  - all classes that you created while implementing the Decorator design pattern :heavy_check_mark:

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      @person.age.should eql 29
+    end
+  end
+
+  context 'Check person parent permission' do
+    it 'return the person permission ' do
+      expect(@person.can_use_services?).to be true
+    end
+  end
+end
```
Comment (AnselemOdims) [inline spec/person_spec.rb:17]: - [x] Please include tests for all the methods of the Person class. For instance, the tests for `add_rental` and `validate_name` are missing

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      @book.title.should eql 'title'
+    end
+  end
+
+  context 'checking the author' do
+    it 'returns the rigth author' do
+      @book.author.should eql 'author'
+    end
+  end
+end
```
Comment (AnselemOdims) [inline spec/book_spec.rb:13]: - [x] Please kindly write tests for the `rent` method on this class

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      @student.age.should eql 28
+    end
+  end
+
+  context 'Check student parent permission' do
+    it 'return the student permission ' do
+      expect(@student.can_use_services?).to be true
+    end
+  end
+end
```
Comment (AnselemOdims) [inline spec/student_spec.rb:15]: - [x] Please kindly write tests for the `play_hooky` method and also the `classroom` setter method

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Juanse7793 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KDlamini — APPROVED** [review]
### Hi team @Juanse7793, @ivaan1992 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the OOP school library: unit tests as per project requirements. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  Unit tests are implemented correctly
✔️  Correct PR title and description
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**